### PR TITLE
[Fix #10216] Update the release documentation with feedback from the 21.7.0 release.

### LINKED ISSUE #10216: Update release documentation 2021.07
|[adiroiban's avatar](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#10216|
|Type|enhancement|
|Created|2021-07-08 16:55:14Z|

Based on the feedback from 21.7.0

<details><summary>Searchable metadata</summary>

```
trac-id__10216 10216
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1625763314506090 1625763314506090
changetime__1646513410089283 1646513410089283
version__None None
owner__None None

```
</details>

### PR DESCRIPTION
**Pull request by adiroiban**
## Scope and purpose

This is a minor release doc update  while doing an actual release.

Updated docs at https://twisted--1616.org.readthedocs.build/en/1616/core/development/policy/release-process.html

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10216
* [NA] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] The title of the PR starts with the associated Trac ticket number (without the `#` character).
* [NA] I have updated the automated tests and checked that all checks for the PR are green.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.
* [x] The merge commit will use the below format
  The first line is automatically generated by GitHub based on PR ID and branch name.
  The other lines generated by GitHub should be replaced.

```
Merge pull request #123 from twisted/4356-branch-name-with-trac-id

Author: adiroiban
Reviewer: 
Fixes: ticket:10216

Update the release documentation.
```

### CODE REVIEW

**Comment by adiroiban** [comment]
The main changes is making it clear that a review for a release candidate should be asked only after the RC is published on PyPI,

But maybe we should ask a review before pushing a PR.

Also, before starting the release branch, a message can be sent to IRC or mailing list to see if there is any PR for which the release can be delayed with 1 or 2 days... but not longer.
But maybe we don't need to document this and any release manager can do it or not.

Or just a info that as an optional step, the release manager can communicate the intention of a release .

------------

Quick link for the review https://twisted--1616.org.readthedocs.build/en/1616/core/development/policy/release-process.html

**Review by graingert — APPROVED** [review]

File: docs/core/development/policy/release-process.rst [hunk 1]
```diff
@@ -13,7 +13,7 @@ Outcomes
 By the end of a Twisted release we'll have:
 
 - Wheel and sdist package published on `PyPI Twisted project <https://pypi.org/project/Twisted/>`_.
-- Updated documentation (API & howtos) on `Twisted Read The Docs <https://twisted.readthedocs.io/en/latest/>`_
+- Updated documentation (API & howtos) on `Twisted Read The Docs <https://https://docs.twistedmatrix.com/>`_
```
Comment (graingert) [inline docs/core/development/policy/release-process.rst]:
```suggestion
- Updated documentation (API & howtos) on `Twisted Read The Docs <https://docs.twistedmatrix.com/>`_
```

**Review by graingert — CHANGES_REQUESTED** [review]
broken URL scheme

**Comment by adiroiban** [comment]
Thanks @graingert for the review. Much appreciated.

I am still updating these notes while working on the release.

I will send one final review request after the final Twisted 21.7.0 is out.

**Comment by adiroiban** [comment]
This is ready for a final review.
I left my comments based on the latest release experience.

For now, read the docs updates are manual ... I hope it can be fixed in https://github.com/twisted/twisted/pull/1629
but the documentation should for now mention that after  the release you need to manually create the documentation for the tag.

File: docs/core/development/policy/release-process.rst [hunk 2]
```diff
@@ -69,28 +86,38 @@ To release Twisted, we
...
    - ``git checkout origin/trunk``
-   - ``git checkout -b release-$RELEASE-4290``
+   - ``git checkout -b release-21.7.0-4290``
 
 
 How to do a release candidate
 -----------------------------
 
 
+This section described the required to create the first release candidate.
```
Comment (graingert) [inline docs/core/development/policy/release-process.rst]: I don't understand this line? A word missing?

File: docs/core/development/policy/release-process.rst [hunk 3]
```diff
@@ -177,29 +236,39 @@ Announce
 
 #. Announce the release
 
-   - Send a text version of the announcement to: twisted-python@twistedmatrix.com, python-announce-list@python.org, twisted-web@twistedmatrix.com
-   - ​http://labs.twistedmatrix.com (Post a web version of the announcements, with links instead of literal URLs)
-   - Twitter, if you feel like it
-   - ``#twisted`` topic on IRC (you'll need ops)
+   - Send a text version of the announcement to: twisted-python@twistedmatrix.com
```
Comment (graingert) [inline docs/core/development/policy/release-process.rst]:
```suggestion
   - Send a text version of the announcement to: twisted@python.com
```

File: docs/core/development/policy/release-process.rst (hunk 2 above)
Comment (adiroiban) [reply docs/core/development/policy/release-process.rst]:
```suggestion
This section describes the steps and requirements for creating the first release candidate.
```

Comment (graingert) [reply docs/core/development/policy/release-process.rst]: Nice

Comment (twm) [reply docs/core/development/policy/release-process.rst]: Yah that's it, "required to create" tripped me up.

File: docs/core/development/policy/release-process.rst [hunk 4]
```diff
@@ -236,7 +236,7 @@ Announce
 
 #. Announce the release
 
-   - Send a text version of the announcement to: twisted-python@twistedmatrix.com
+   - Send a text version of the announcement to: twisted@python.com
```
Comment (graingert) [inline docs/core/development/policy/release-process.rst]:
```suggestion
   - Send a text version of the announcement to: twisted@python.org
```

Apologies I got this wrong :/

**Review by graingert — APPROVED** [review]

**Pull request merged**